daffodil-tdml/src/main/scala/org/apache/daffodil/tdml/TDMLRunner.scala

```diff
@@ -275,16 +284,17 @@ class DFDLTestSuite private[tdml] (
   // We call it an UnparserTestCase
   //
   val unparserTestCases = (ts \ "unparserTestCase").map { node => UnparserTestCase(node, this) }
-  val testCases: Seq[TestCase] = parserTestCases ++
-    unparserTestCases
-  val duplicateTestCases = testCases.groupBy { _.name }.filter { case (name, seq) => seq.length > 1 }
-  if (duplicateTestCases.size > 0) {
+  val testCases: Seq[TestCase] = parserTestCases ++ unparserTestCases
+  val duplicateTestCases = testCases.groupBy {
+    _.tcName
+  }.filter { case (name, seq) => seq.length > 1 }
+  if (duplicateTestCases.nonEmpty) {
     duplicateTestCases.foreach {
       case (name, _) =>
         System.err.println("TDML Runner: More than one test case for name '%s'.".format(name))
```

Comment:
It is tolerated because we didn't want to hunt them down and fix them. They were there in the test suite, quite a few, at one time. I haven't checked lately but there's a JIRA ticket to fix them I think, at which point yes, this should be an exception.